static/app/utils/fields/index.ts

```diff
@@ -308,128 +310,443 @@ export interface FieldDefinition {
   values?: string[];
 }
 
+type ColumnValidator = (field: {key: string; valueType: FieldValueType}) => boolean;
+
+function validateForNumericAggregate(
+  validColumnTypes: FieldValueType[]
+): ColumnValidator {
+  return function ({key, valueType}) {
+    // these built-in columns cannot be applied to numeric aggregates such as percentile(...)
+    if (
+      [
+        FieldKey.DEVICE_BATTERY_LEVEL,
+        FieldKey.STACK_COLNO,
+        FieldKey.STACK_LINENO,
+        FieldKey.STACK_STACK_LEVEL,
+      ].includes(key as FieldKey)
+    ) {
+      return false;
+    }
+
+    return validColumnTypes.includes(valueType);
+  };
+}
+
+function validateAndDenyListColumns(
+  validColumnTypes: FieldValueType[],
+  deniedColumns: string[]
+): ColumnValidator {
+  return function ({key, valueType}) {
+    return validColumnTypes.includes(valueType) && !deniedColumns.includes(key);
+  };
+}
+
+function validateAllowedColumns(validColumns: string[]): ColumnValidator {
+  return function ({key}): boolean {
+    return validColumns.includes(key);
+  };
+}
+
 export const AGGREGATION_FIELDS: Record<AggregationKey, FieldDefinition> = {
   [AggregationKey.COUNT]: {
     desc: t('count of events'),
     kind: FieldKind.FUNCTION,
     valueType: FieldValueType.NUMBER,
+    parameters: [],
   },
   [AggregationKey.COUNT_UNIQUE]: {
     desc: t('Unique count of the field values'),
     kind: FieldKind.FUNCTION,
     valueType: FieldValueType.INTEGER,
+    parameters: [
+      {
+        name: 'column',
+        kind: 'column',
+        columnTypes: [
+          FieldValueType.STRING,
+          FieldValueType.DURATION,
+          FieldValueType.NUMBER,
+          FieldValueType.INTEGER,
+          FieldValueType.DURATION,
+          FieldValueType.BOOLEAN,
+        ],
+        defaultValue: 'user',
+        required: true,
+      },
+    ],
   },
   [AggregationKey.COUNT_MISERABLE]: {
     desc: t('Count of unique miserable users'),
     kind: FieldKind.FUNCTION,
+    parameters: [
+      {
+        name: 'column',
+        kind: 'column',
+        columnTypes: validateAllowedColumns(['user']),
+        defaultValue: 'user',
+        required: true,
+      },
+      {
+        name: 'value',
+        kind: 'value',
+        dataType: FieldValueType.NUMBER,
+        defaultValue: '300',
+        required: true,
+      },
+    ],
     valueType: FieldValueType.NUMBER,
   },
   [AggregationKey.COUNT_IF]: {
     desc: t('Count of events matching the parameter conditions'),
     kind: FieldKind.FUNCTION,
     valueType: FieldValueType.NUMBER,
+    parameters: [
+      {
+        name: 'column',
+        kind: 'column',
+        columnTypes: validateAndDenyListColumns(
```

Comment:
Does this function need to filter out aggregate keys, or should we modify SearchQueryBuilder to automatically do that?

![CleanShot 2024-08-14 at 11 36 14](https://github.com/user-attachments/assets/0c3abcf6-df22-43a3-b2b5-0921f99eba89)

Reply:
@malwilley can you elaborate a bit on how we can `modify SearchQueryBuilder to automatically` to ensure the valueType of the column is in a certain list and exclude some columns with in a certain list.

Reply:
Yeah! So the two options are:

1. `validateAndDenyListColumns` could return null for things like `count`, `count_if`, etc
2. `SearchQueryBuilder` could filter out all aggregate filter keys before passing to the `columnTypes` function

Option 1 is more flexible, but option 2 would be good if we _never_ want aggregate keys to be suggested as parameter values.

Reply:
Ahh I see @malwilley , I misunderstood the screen shot. Definitely go for option 2. We never expect aggregates as parameters.

Reply:
Great, I will also follow up with that change!